metron-interface/metron-config/e2e/sensor-config/sensor-config.po.ts

```diff
@@ -0,0 +1,243 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { browser, element, by, protractor } from 'protractor/globals';
+import {changeURL, waitForElementVisibility, waitForElementInVisibility, waitForElementPresence} from '../utils/e2e_util';
+
+export class SensorConfigPage {
+
+  _getEnrichmentsMultipleInput() {
+    return element.all(by.css('.config.container')).all(by.css('metron-config-multiple-input')).get(1).all(by.css('select')).last();
+  }
+
+  _getThreatIntelMultipleInput() {
+    return element.all(by.css('.config.container')).all(by.css('metron-config-multiple-input')).last().all(by.css('select')).last();
+  }
+
+  _getTransformationMultipleInput() {
+    return element.all(by.css('.config.container')).all(by.css('metron-config-multiple-input')).first().all(by.css('select')).last();
+  }
+
+  clickAddButton() {
+    changeURL(browser.baseUrl + '/sensors');
+    let addButton = element(by.css('.metron-add-button.hexa-button'));
+    return waitForElementPresence(addButton).then(() => {
+      addButton.click();
+    });
+  }
+
+  clickAddThreatTriageRule() {
+    let addThreatTriageButton = element(by.css('metron-config-sensor-threat-triage .add-button'));
+    return waitForElementVisibility(addThreatTriageButton).then(() => {
+      return addThreatTriageButton.click();
+    });
+  }
+
+  clickGrokStatement() {
+    let grokInput = element(by.css('input[formcontrolname="grokStatement"] + span'));
+    return waitForElementVisibility(grokInput).then(() => {
+      return grokInput.click();
+    });
+  }
+
+  clickSchema() {
+    let schemaInput = element(by.css('div[name="fieldSchema"] button'));
+    return waitForElementVisibility(schemaInput).then(() => {
+      return schemaInput.click();
+    });
+  }
+
+  clickThreatTriage() {
+    let threatTriageInput = element(by.css('div[name="threatTriage"] button'));
+    return waitForElementVisibility(threatTriageInput).then(() => {
+      return threatTriageInput.click();
+    });
+  }
+
+  closeMainPane() {
+    return element(by.css('.btn.save-button + button')).click();
+  }
+
+  closeSchemaPane() {
+    return element.all(by.css('metron-config-sensor-field-schema .form-title + i')).click();
+  }
+
+  closeThreatTriagePane() {
+    return element.all(by.css('metron-config-sensor-threat-triage .form-title + i')).click();
+  }
+
+  getGrokStatementFromMainPane() {
+    browser.waitForAngular;
+    return element.all(by.css('input[formcontrolname="grokStatement"]')). getAttribute('value');
+  }
+
+  getFieldSchemaSummary() {
+    return element.all(by.css('[name="fieldSchema"] table tr')).getText();
+  }
+
+  getFieldSchemaValues() {
+    return waitForElementPresence(element(by.css('metron-config-sensor-field-schema .field-schema-row'))).then(() => {
+      return element.all(by.css('metron-config-sensor-field-schema .field-schema-row')).getText();
+    });
+  }
+
+  getGrokResponse() {
```

Comment:
This seems surprisingly sensor specific given the generality of the other calls. Are we presuming too much about the type of sensor here?